api/src/test/java/com/epam/pipeline/controller/contextual/ContextualPreferenceControllerTest.java

```diff
@@ -0,0 +1,219 @@
+ */
+
+package com.epam.pipeline.controller.contextual;
+
+import com.epam.pipeline.controller.ResponseResult;
+import com.epam.pipeline.controller.Result;
+import com.epam.pipeline.controller.vo.ContextualPreferenceVO;
+import com.epam.pipeline.entity.contextual.ContextualPreference;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceExternalResource;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceLevel;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceSearchRequest;
+import com.epam.pipeline.manager.contextual.ContextualPreferenceApiService;
+import com.epam.pipeline.test.creator.contextual.ContextualPreferenceCreatorUtils;
+import com.epam.pipeline.test.web.AbstractControllerTest;
+import com.epam.pipeline.util.ControllerTestUtils;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = ContextualPreferenceController.class)
+public class ContextualPreferenceControllerTest extends AbstractControllerTest {
+
+    private static final String TEST_STRING = "TEST";
+    private static final String CONTEXTUAL_URL = SERVLET_PATH + "/contextual/preference";
+    private static final String LOAD_URL = CONTEXTUAL_URL + "/load";
+    private static final String LOAD_ALL_URL = LOAD_URL + "/all";
+    private static final ContextualPreferenceLevel PREFERENCE_LEVEL = ContextualPreferenceLevel.ROLE;
+    private ContextualPreference contextualPreference;
+    private ResponseResult<ContextualPreference> expectedResult;
+    private ContextualPreferenceExternalResource contextualPreferenceExternalResource;
+
+    @Autowired
+    private ContextualPreferenceApiService mockContextualPreferenceApiService;
+
+    @Before
+    public void setUp() {
+        contextualPreference = ContextualPreferenceCreatorUtils.getContextualPreference();
+        expectedResult = ControllerTestUtils.buildExpectedResult(contextualPreference);
+
+        contextualPreferenceExternalResource
+                = ContextualPreferenceCreatorUtils.getCPExternalResource();
+    }
```

Comment:
It seems cleaner to inline test class field initiliazations and get rid of `setUp` method.